keyboards/handwired/co60/keymaps/jmdaly_hhkb_split_space/keymap.c

```diff
@@ -0,0 +1,160 @@
+
+enum co60_layers {
+  _L1,
+  _L2,
+  _L3
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  // Default base layer. Except for the bottom row, this
+  // is close to a standard HHKB layout.
+  [_L1] = LAYOUT_60_hhkb_split_625u_space( /* Base */
+    KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7, KC_8,    KC_9,    KC_0,    KC_MINS,    KC_EQL,  KC_BSLS, KC_GRV, \
+    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U, KC_I,    KC_O,    KC_P,    KC_LBRC,    KC_RBRC, KC_BSPC, \
+    CTL_T(KC_ESC), KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J, KC_K,    KC_L,    KC_SCLN, KC_QUOT,    KC_ENT,  \
+    KC_LSPO, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M, KC_COMM, KC_DOT,  KC_SLSH, KC_RSPC, MO(_L3),   \
+             KC_LALT, KC_LGUI,       KC_ENT,              MO(_L3),       KC_SPC,  KC_LEAD, KC_RGUI, KC_RALT \
+  ),
+  // My preferred base layout. This doesn't match the caps
+  // on my boards, so I don't make it default.
+  [_L2] = LAYOUT_60_hhkb_split_625u_space( /* Base */
+    KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7, KC_8,    KC_9,    KC_0,    KC_MINS,    KC_EQL,  KC_BSLS, KC_DEL, \
+    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U, KC_I,    KC_O,    KC_P,    KC_LBRC,    KC_RBRC, KC_BSPC, \
+    CTL_T(KC_ESC), KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J, KC_K,    KC_L,    KC_SCLN, KC_QUOT,    KC_ENT,  \
+    KC_LSPO, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M, KC_COMM, KC_DOT,  KC_SLSH, KC_RSPC, MO(_L3),   \
+             KC_LALT, KC_LGUI,       KC_ENT,              MO(_L3),       KC_SPC,  KC_LEAD, KC_RGUI, KC_RALT \
+  ),
+  [_L3] = LAYOUT_60_hhkb_split_625u_space( /* Function */
+    RESET,   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,     KC_F10,    KC_F11,     KC_F12,   _______, _______,   \
+    BL_BRTG, BL_TOGG, BL_INC,  BL_DEC, BL_ON, BL_OFF, _______, _______, _______, _______,   _______,   KC_PGUP,    KC_INSERT,  KC_DEL, \
+    _______, RGB_TOG, RGB_MOD, RGB_RMOD, _______, _______, KC_LEFT, KC_DOWN, KC_UP,   KC_RIGHT,  KC_HOME,   KC_END,     _______,  \
+    _______, BL_DEC,  _______, _______, _______, _______, _______, DF(_L1), DF(_L2), _______,   KC_PGDOWN, _______, _______,   \
+             _______, _______,          _______,          _______,          _______, _______,   _______,   _______ \
+  )
+};
+
+const uint16_t PROGMEM fn_actions[] = {
+
+};
+
+const macro_t *action_get_macro(keyrecord_t *record, uint8_t id, uint8_t opt)
+{
+  // MACRODOWN only works in this function
+      switch(id) {
+        case 0:
+          if (record->event.pressed) {
+            register_code(KC_RSFT);
+          } else {
+            unregister_code(KC_RSFT);
+          }
+        break;
+      }
+    return MACRO_NONE;
+};
+
+
+void matrix_init_user(void) {
+
+}
+
+LEADER_EXTERNS();
+
+void matrix_scan_user(void) {
+
+  LEADER_DICTIONARY() {
```

Comment:
I'm not sure `matrix_scan_user()` is the right place for this Leader key stuff - it will be run every time the matrix is scanned (many, many times a second).

Reply:
Yes, it is.

https://docs.qmk.fm/#/feature_leader_key

Reply:
Ah, should have checked. Never mind.